apm-agent-core/src/main/java/co/elastic/apm/agent/report/serialize/DslJsonSerializer.java

```diff
@@ -693,7 +693,11 @@ private static boolean isExcluded(StackTraceElement stackTraceElement) {
 
     private void serializeStackTraceElement(StackTraceElement stacktrace) {
         jw.writeByte(OBJECT_START);
-        writeField("filename", stacktrace.getFileName());
+        if (apmServerClient.supportsStacktraceClassname()) {
+            writeField("classname", stacktrace.getClassName());
+        } else {
+            writeField("filename", stacktrace.getFileName());
+        }
```

Comment:
As per https://github.com/elastic/kibana/issues/49467#issuecomment-571738211 - let's send both for a while.